libcontainer/container_linux.go

```diff
@@ -460,7 +461,42 @@ func (c *linuxContainer) newParentProcess(p *Process) (parentProcess, error) {
 	return c.newInitProcess(p, cmd, parentPipe, childPipe)
 }
 
+func runcInTmp() (string, error) {
+	runc, err := os.Readlink("/proc/self/exe")
+	if err != nil {
+		return "", err
+	}
+	id := uuid.New()
+	temp := fmt.Sprintf("/tmp/runc.%s", id)
```

Comment:
If we're going to do it this way, I would want to use `memfd_create` because using `/tmp` won't always work and `memfd_create` is a fool-proof way of doing it. But ultimately I could just rewrite #1984 entirely to not use C code.